spec/rollbar/plugins/rails_js_spec.rb

```diff
@@ -20,37 +35,65 @@
   end
 
   context 'using no security policy' do
+    before(:all) do
+      reset_secure_headers_config # disable secure_headers gem
+    end
+
     include_examples 'adds the snippet'
   end
 
   context 'using rails5 content_security_policy',
           :if => (Gem::Version.new(Rails.version) >= Gem::Version.new('5.2.0')) do
     def configure_csp(mode)
-      Rails.application.config.content_security_policy_nonce_generator = lambda { |_| SecureRandom.base64(16) }
       if mode == :nonce_present
         nonce_present
+      elsif mode == :nonce_not_present
+        nonce_not_present
       elsif mode == :script_src_not_present
         script_src_not_present
+      elsif mode == :unsafe_inline_present
+        unsafe_inline_present
       else
         raise 'Unknown CSP mode'
       end
     end
 
     def nonce_present
-      # Rails will add the nonce to script_src automatically, when script_src is present.
+      # Rails will add the nonce to script_src when the generator is set.
+      Rails.application.config.content_security_policy_nonce_generator = lambda { |_| SecureRandom.base64(16) }
+
+      Rails.application.config.content_security_policy do |policy|
+        policy.script_src :self, :https
+      end
+    end
+
+    def nonce_not_present
+      Rails.application.config.content_security_policy_nonce_generator = nil
+
       Rails.application.config.content_security_policy do |policy|
         policy.script_src :self, :https
       end
```

Comment:
It seems like some functions (here and below) have redundant definition and it might be wise to extract some methods. However, as it's spec file it's probably better to leave it in the current state, so I only marking it for taking under consideration.

Reply:
I might be misunderstanding what you're looking at, but while the method names are the same in both places, the implementation is different for Rails 5 content security policy vs the secure headers gem that is tested in the other case.

Reply:
I'm not talking about method names but the implementation of the methods.

The only difference between `nonce_present` and `nonce_not_present` methods is this:

`Rails.application.config.content_security_policy_nonce_generator` = `lambda { |_| SecureRandom.base64(16) }` OR `nil`

The rest of the code is equivalent:
```ruby
     Rails.application.config.content_security_policy do |policy|
        policy.script_src :self, :https
     end
```

Similar pattern happens to other functions below this snippet.

Reply:
@waltjones just for clarification. I don't think there is anything wrong and/or needing refactoring here. As long as it was intended to duplicate same logic per test, it's okay to keep redundant lines as this simplify and isolate tests. I only wanted to raise your attention in case this was unintended.

Reply:
:+1: No worries.

Reply:
@bxsx Apologies, it took me awhile to get back to respond to your comment.

It's intentional. I'd say it's merely a coincidence that two of the current test cases have the same directive list for now. (And a third one doesn't.) For the shared method to remain useful in the future, it would need to accept an arbitrary set of directives. `Rails.application.config.content_security_policy` already does this, but wrapping it could still be valuable. If that method changes in a future Rails version, we only want to branch on Rails version in one place. So for a wrapper, we definitely want to accept an arbitrary set of directives, so all of the current test cases can use it, and any that are added later that also have different directives. (Or if one of the current test cases changes its directives.)

As the code stands, we'd be replacing a method that accepts a block with a wrapper method that accepts a block. Not much code reduction, but still valuable in the future if we need to branch on different Rails versions.

If this were used in more than a few places, I might go ahead and do that now. Maybe? We'd just be betting on this part of Rails changing at some point and most likely in this case it won't.

FWIW, I'd probably say the same if it were in non-test code, assuming there was the same volatility in what directives would be needed each time, it's already a single method call, and it's used a handful of times or less.